### PR TITLE
Add date to deployment

### PR DESCRIPTION
**Pull request by laaksonm**

### CODE REVIEW

File: scripts/deploy [hunk 1]
```diff
@@ -60,14 +61,9 @@ def bump_version(current_version: Version, bump_part: str) -> Version:
...
     ]
     if bump_part in DEFAULT_SPECIAL_RELEASE_LETTER:
-        special_version_number = getattr(current_version, bump_part)
-        # only pre-release version return tuples, others return number
-        if special_version_number and bump_part == 'pre':
-            special_version_number = special_version_number[1]
-        if not special_version_number:
-            parts.append(f'{DEFAULT_SPECIAL_RELEASE_LETTER[bump_part]}1')
-        else:
-            parts.append(f'{DEFAULT_SPECIAL_RELEASE_LETTER[bump_part]}{special_version_number + 1}')
```
Comment (tringm) [inline scripts/deploy]: This part is to retrieve the current version number and bump by 1. If we remove this part, you can no longer bump version for post, pre (rc), and dev release.

Comment (tringm) [reply scripts/deploy]: Instead, you should add an option flag, for instance:
```
self.add_argument(
            '-d', '--append-datetime', type=bool, action='store_true', help='append datetime at the end of version, only applicable for dev'
        )
```
Then something like
```
special_version_number = getattr(current_version, bump_part)
        # only pre-release version return tuples, others return number
        if special_version_number and bump_part == 'pre':
            special_version_number = special_version_number[1]
       if use_datetime and bump_part == 'dev':
            parts.append(f'dev{datetime.utcnow()})
       elif not special_version_number:
            parts.append(f'{DEFAULT_SPECIAL_RELEASE_LETTER[bump_part]}1')
        else:
            parts.append(f'{DEFAULT_SPECIAL_RELEASE_LETTER[bump_part]}{special_version_number + 1}')
```

Comment (tringm) [reply scripts/deploy]: A better way is handling the flag at parser, then use a seprate function from bump version where you check, for instance, that you can only append date with dev version.

File: .circleci/config.yml [hunk 2]
```diff
@@ -303,7 +303,7 @@ commonYAMLStructure:
       name: deploy to test pypi
```
Comment (tringm) [inline .circleci/config.yml]:
```suggestion
      name: build and install package
```
It was a wrong name

File: .circleci/config.yml [hunk 3]
```diff
@@ -303,7 +303,7 @@ commonYAMLStructure:
...
       command: |
         . venv/bin/activate
-        ./scripts/deploy test.pypi --no-publish
+        ./scripts/deploy test.pypi --no-publish --bump-version dev
```
Comment (tringm) [inline .circleci/config.yml:306]:
```suggestion
        ./scripts/deploy test.pypi --no-publish
```
This does not need to be changed. We just build package locally

**Review by tringm — APPROVED** [review]

File: .circleci/config.yml (hunk 3 above)
Comment (laaksonm) [reply .circleci/config.yml:306]: The script has the required for either version or bump-version, so it has to be there or the script won't run.

**Pull request merged**